framework/src/base/Adaptivity.C

```diff
@@ -214,6 +232,22 @@ Adaptivity::adaptMesh(std::string marker_name /*=std::string()*/)
       _displaced_mesh_refinement->flag_elements_by_error_fraction(*_error);
   }
 
+  // Moving some of h flagged elements to p flagged based on the
+  // local smoothness and prior h & p error estimates
+  if (_adaptivity_type == AdaptivityType::HP)
+  {
+    if (_displaced_problem)
+      mooseError("HP refinement cannot currently be used with a displaced mesh. This is because "
+                 "we may toggle from h-refinement to p-refinement based on solution, gradient "
+                 "(for C0 elements), and hessian (for C0 and C1 elements) differences between "
+                 "coarse and fine states, and the gradient and hessian will be functions of mesh "
+                 "displacements. Consequently it's possible that refinement flags may end up out "
+                 "of sync between reference and displaced meshes which would be disastrous");
```

Comment:
Wait, surely we're not just throwing the same error estimators at each mesh independently for h or p refinement?  Do we not do our refinement on a single mesh and then copy flags over to the other?